Core/MaterialExtension.h

```diff
@@ -31,60 +33,38 @@
 class MaterialExtension : public G4Material
 {
 public:
-  MaterialExtension(const G4String& name, const G4Material* baseMaterial);
+  MaterialExtension(const G4String& materialType, const G4String& name, const G4Material* baseMaterial);
   ~MaterialExtension();
 
-  G4Material* GetMaterial()
-  {
-    return fMaterial;
-  };
-
-  void SetoPsLifetime(G4double);
-
-  void SetoPsFraction(G4double);
-
-  void SetPickOffFraction(G4double sf)
-  {
-    fPickOffFraction = sf;
-  }
-
-  const G4double Get3gFraction();
-
-  const G4double GetoPsLifetime()
-  {
-    return foPslifetime;
-  };
-
+  G4Material* GetMaterial(){return fMaterial;};
+  void Add_oPsComponent(G4double Lifetime, G4double Probability);
+  void Add_directComponent(G4double Lifetime, G4double Probability);
+  void Set_pPsComponent(G4double Lifetime, G4double Fraction);
+  const G4double GetLifetime(double randNumber, G4String channel) const;
```

Comment:
I would suggest that you use an enum type for `channel` rather than a string to avoid typos in the later usages of this method.

Reply:
I agree, changed to enum